ykpack/src/types.rs

```diff
@@ -323,15 +323,22 @@ impl Display for SignedInt {
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
 pub enum CallOperand {
     /// A statically known function identified by its DefId.
-    Fn(DefId),
+    /// A pair: the definition ID and the binary symbol name (if known).
+    Fn(DefId, Option<String>),
     /// An unknown or unhandled callable.
     Unknown, // FIXME -- Find out what else. Closures jump to mind.
 }
 
 impl Display for CallOperand {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            CallOperand::Fn(def_id) => write!(f, "{}", def_id),
+            CallOperand::Fn(def_id, sym_name) => {
+                let sym_name_str = match sym_name {
+                    Some(n) => n,
+                    None => "<unknown>",
```

Comment:
What does it mean for it to be unknown?

Reply:
If our compiler encounters a MIR body which isn't full instantiated (i.e. still has "free" type parameters), then it won't be code-genned, thus it won't have a symbol name.

I have a comment in the compiler about this, but I suppose I should say that here too.

Reply:
A brief comment might be worthwhile, but I was more curious as to whether there was something deeper going on or not.